src/playground/sample_data_tooling/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,345 @@
+
+    Raises:
+        - ValueError: current_index should be a positive index position
+        - TypeError: dataset is a list of documents to ingest
+    """
+
+    # First validates input
+    ingest_validation(index_name = index_name,
+        file_provided = file_provided,
+        timestamp = timestamp,
+        minutes = minutes,
+        chunk = chunk,
+        max_bulk_size = max_bulk_size
+    )
+    # Validates function specific input
+    if type(current_index) is not int or (type(current_index) is int and current_index < 0):
+        raise ValueError("current_index should be a positive index position")
+    if type(dataset) is not list:
+        raise TypeError("dataset is a list of documents to ingest")
+
+    start_index = current_index
+    one_week_ago = datetime.now() - timedelta(days = 7)
+    add_to_request = []
+    request_body = []
+    current_body_size = sys.getsizeof(request_body)
+    while current_index < min(start_index + chunk, len(dataset)):
+        # Adds the index name
+        index_name_body = {"index": {"_index": index_name}}
+
+        # If timestamps were a field type for user provided data, modify time
+        if file_provided and timestamp:
+            one_week_ago += timedelta(minutes = minutes)
+            offset = one_week_ago.strftime("%s")
+            dataset[current_index][timestamp] = int(offset)
+
+        # Adds the action to take
+        index_action = dataset[current_index]
+
+        # Adds both index name body and index action body to the test request body
+        add_to_request.append(index_name_body)
+        add_to_request.append(index_action)
+        current_body_size = sys.getsizeof(add_to_request)
+
+        # Adds to request only if the body size is < the max bulk size (user defined)
+        if current_body_size < max_bulk_size:
+            request_body.append(index_name_body)
+            request_body.append(index_action)
+        else:
+            break
+        current_index+= 1
+
+    return (request_body, current_index)
+
+
+def ingest(client:OpenSearch,
+    data_template,
+    index_name:str,
+    file_provided:bool = False,
+    mapping:bool = True,
+    number:int = 6,
+    chunk:int = 5,
+    timestamp:str = None,
+    minutes:int = 2,
+    current_date:datetime = datetime.today(),
+    max_bulk_size:int = 100000,
+    anomaly_detection_trend:dict = None
+) -> list:
+    """
+    Function that ingests user-provided data or generated data
+
+    Arguments:
+        - client: an OpenSearch Python client object
+        - data_template: JSON file, CSV file, JSON mapping, or JSON short-hand template
+        - index_name: The name of the index to ingest data
+        - mapping: Whether or not the input is a mapping
+        - file_provided: Boolean flag as to whether or not a file was provided
+        - chunk: How many documents can be ingested per BULK call
+        - timestamp: The field name that contains timestamps
+        - number: How many documents to generate
```

Comment:
you provides the default value 6, can you add the information here? Thanks

Reply:
Yes, I've added default value information to all the args.